client/internal/bootstrap/client_test.go

```diff
@@ -23,382 +24,359 @@ import (
 	akssecuretlsbootstrapv1_mocks "github.com/Azure/aks-secure-tls-bootstrap/service/pkg/gen/mocks/akssecuretlsbootstrap/v1"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/assert"
 	"go.uber.org/mock/gomock"
 	"go.uber.org/zap"
 )
 
 // TODO: refactor towards vanilla go tests
 
-var _ = Describe("Client tests", Ordered, func() {
+func setupConfig(t *testing.T) *Config {
+	t.Helper()
+
 	const (
 		apiServerFQDN  = "controlplane.azmk8s.io"
 		kubeconfigPath = "path/to/kubeconfig"
 	)
-	var (
-		ctx                 context.Context
-		mockCtrl            *gomock.Controller
-		imdsClient          *imdsmocks.MockClient
-		kubeconfigValidator *kubeconfigmocks.MockValidator
-		serviceClient       *akssecuretlsbootstrapv1_mocks.MockSecureTLSBootstrapServiceClient
-		bootstrapClient     *Client
-		bootstrapConfig     *Config
-		logger              *zap.Logger
-
-		clusterCAFilePath string
-		credFilePath      string
+
+	clusterCACertPEM, _, err := testutil.GenerateCertPEM(
+		testutil.CertTemplate{
+			CommonName:   "hcp",
+			Organization: "aks",
+			IsCA:         true,
+			Expiration:   time.Now().Add(time.Hour),
+		},
 	)
+	assert.NoError(t, err)
+
+	tempDir := t.TempDir()
+	clusterCAFilePath := filepath.Join(tempDir, "ca.crt")
+	credFilePath := filepath.Join(tempDir, "client.pem")
+
+	err = os.WriteFile(clusterCAFilePath, clusterCACertPEM, os.ModePerm)
+	assert.NoError(t, err)
+
+	return &Config{
+		NextProto:         "bootstrap",
+		AADResource:       "resource",
+		ClusterCAFilePath: clusterCAFilePath,
+		CredFilePath:      credFilePath,
+		APIServerFQDN:     apiServerFQDN,
+		KubeconfigPath:    kubeconfigPath,
+		ProviderConfig: cloud.ProviderConfig{
+			CloudName:    azure.PublicCloud.Name,
+			ClientID:     "service-principal-id",
+			ClientSecret: "service-principal-secret",
+			TenantID:     "tenantId",
+		},
+	}
+}
 
-	BeforeAll(func() {
-		logger, _ = zap.NewDevelopment()
-
-		clusterCACertPEM, _, err := testutil.GenerateCertPEM(
-			testutil.CertTemplate{
-				CommonName:   "hcp",
-				Organization: "aks",
-				IsCA:         true,
-				Expiration:   time.Now().Add(time.Hour),
-			},
-		)
-		Expect(err).To(BeNil())
-
-		tempDir := GinkgoT().TempDir()
-		clusterCAFilePath = filepath.Join(tempDir, "ca.crt")
-		credFilePath = filepath.Join(tempDir, "client.pem")
-
-		err = os.WriteFile(clusterCAFilePath, clusterCACertPEM, os.ModePerm)
-		Expect(err).To(BeNil())
-	})
+func setupBootstrapClientTestDeps(t *testing.T, logger *zap.Logger) (*Client, context.Context,
```

Comment:
again, we can refactor all the t.Run calls to table format, and have the test loop perform all the stuff you're doing in this func